src/utils/breakpoint/index.js

```diff
@@ -0,0 +1,43 @@
+import { updateObj } from "../utils";
+// Return the first argument that is a string, or null if nothing is a
+// string.
+export function firstString(...args) {
+  for (let arg of args) {
+    if (typeof arg === "string") {
+      return arg;
+    }
+  }
+  return null;
+}
+
+export function locationMoved(location, newLocation) {
+  return (
+    location.line !== newLocation.line ||
+    (location.column != null && location.column !== newLocation.column)
+  );
+}
+
+export function makeLocationId(location: Location) {
+  const { sourceId, line, column } = location;
+  const columnString = column || "";
+  return `${sourceId}:${line}:${columnString}`;
+}
+
+export function makePendingLocationId(location: Location) {
+  const { sourceUrl, line, column } = location;
+  const sourceUrlString = sourceUrl || "";
+  const columnString = column || "";
+  return `${sourceUrlString}:${line}:${columnString}`;
+}
+
+export function allBreakpointsDisabled(state) {
+  return state.breakpoints.every(x => x.disabled);
+}
+
+// syncing
+export function equalizeLocationColumn(location, hasColumn) {
+  if (hasColumn) {
+    return location;
+  }
+  return Object.assign({}, location, { column: undefined });
+}
```

Comment:
2 things:

1. I think we should do this in `firefox/commands.js#onNewBreakpoint`
2. `onNewBreakpoint` should be moved to firefox/create#createBreakpoint

The rationale for this is to give us consistent data from the server... we did the same thing for source URLs when we worked on pending bps